packages/automerge-repo/src/DocHandle.ts

```diff
@@ -448,12 +471,48 @@ export class DocHandle<T> //
   }
 }
 
+function optionsWithGlobalMetadata<T>(
+  options: DocHandleChangeOptions<T>,
+  globalMetadata: ChangeMetadata
+): A.ChangeOptions<T> {
+  let metadata = { ...globalMetadata }
+
+  if (options.metadata) {
+    if (!metadata) {
+      metadata = {}
+    }
+
+    Object.assign(metadata, options.metadata)
+  }
+
+  return {
+    time: options.time,
```

Comment:
I'm assuming this is an epoch? I wouldn't use it unless there was some sort of indication if it's a client or server timestamp, a way to sign client time (trusted timestamp), and a spec for the format.

Reply:
Being able to add timestamps to changes is not a new feature that already existed before. It's purely advisory and it's not used internally for any algorithms.

Reply:
Ok. Sorry that was a very convoluted way to explain myself.
I don't understand what it's for, should it be in the metadata instead?

Reply:
I agree it would be more consistent not to make the timestamp special and instead just put it into the metadata. What do you think @alexjg?

Reply:
Yeah I think the API should be just to have the timestamp in the metadata and then internally pull it out and separately pass it to automerge.

Reply:
Oh, is it used internally somehow? I would actually feel better if it was separate from the metadata then, so we're not hiding a potential complexity.

Reply:
Timestamps are not used by the CRDT logic – they are only informational, to enable things like history visualisation. But the current Automerge data format already has a place for storing timestamps; removing it and moving it into the generic metadata would be a breaking change to the data format.